### PR TITLE
Fix precision loss by removing uses of vecX.fromValues and add related tests

### PR DESCRIPTION
**Pull request by timokorkalainen**
## Launch Checklist

As noticed in issue #1105 zooming caused 3D models to jitter around. This lead to discovery of glMatrix.vecX.fromValues functions to use Float32Arrays instead of Float64Arrays, and thus rounding values unintentionally. The "3D model zooming" case was just a very visible example of how large numbers rounded to integers caused visual glitches.

This PR removes all handful of uses of glMatrix.vecX.fromValues (save for test cases that anyway use whole numbers) and adds unit tests to cases where it was simple to do so.

Before:

https://user-images.githubusercontent.com/1556222/159093273-6f486916-00d1-49f3-ad0a-516b4fb58ba2.mp4

After:

https://user-images.githubusercontent.com/1556222/159093295-4e59d2be-0837-4da5-ba56-ab579ec7c967.mp4

Benchmark scores:
[txt](https://github.com/maplibre/maplibre-gl-js/files/8308125/benchmarks.txt)
[pdf](https://github.com/maplibre/maplibre-gl-js/files/8308124/all.pdf)

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask! (The issue is of Maplibre origin and I have never looked at MapBox v2 source)
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. **Partial: some files would have required very complex tests.**
 - [x] Document any changes to public APIs. **None**
 - [x] Post benchmark scores.
 - [x] Manually test the debug page.
 - [x] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog". **bug**
 - [x] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog>Fix precision loss in some matrix calculations (#1105) </changelog>`.

### CODE REVIEW

File: src/style/style_layer/circle_style_layer.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ class CircleStyleLayer extends StyleLayer {
 }
 
 function projectPoint(p: Point, pixelPosMatrix: mat4) {
-    const point = vec4.transformMat4(vec4.create(), vec4.fromValues(p.x, p.y, 0, 1), pixelPosMatrix);
+    const point = vec4.transformMat4(vec4.create(), [p.x, p.y, 0, 1], pixelPosMatrix);
```
Comment (HarelM) [inline src/style/style_layer/circle_style_layer.ts:88]: Was the vec4.create existed in version 1.x? As it might affect too, right?

**Comment by HarelM** [comment]
Looks good! Thanks!
Please edit the changelog file and add the fixed issue.

File: src/style/style_layer/circle_style_layer.ts (hunk 1 above)
Comment (timokorkalainen) [reply src/style/style_layer/circle_style_layer.ts]: Yes, this was added with the TS conversion too. Reverted this now.

**Comment by timokorkalainen** [comment]
Thanks, issue number added to changelog item.

File: src/style/style_layer/circle_style_layer.ts [hunk 2]
```diff
@@ -85,7 +85,7 @@ class CircleStyleLayer extends StyleLayer {
 }
 
 function projectPoint(p: Point, pixelPosMatrix: mat4) {
-    const point = vec4.transformMat4(vec4.create(), vec4.fromValues(p.x, p.y, 0, 1), pixelPosMatrix);
+    const point = vec4.transformMat4([]as any, [p.x, p.y, 0, 1], pixelPosMatrix);
```
Comment (wipfli) [inline src/style/style_layer/circle_style_layer.ts:88]:
```suggestion
    const point = vec4.transformMat4([] as any, [p.x, p.y, 0, 1], pixelPosMatrix);
```

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
Changelog entry is still missing...

**Comment by timokorkalainen** [comment]
> Changelog entry is still missing...

Ah, sorry, you mean an entry to the actual Changelog file?

**Comment by HarelM** [comment]
Yes, thanks!

**Comment by timokorkalainen** [comment]
Like this?

**Comment by HarelM** [comment]
Yes, thanks!

**Pull request merged**